aws/logs_monitoring/enhanced_lambda_metrics.py

```diff
@@ -254,6 +259,89 @@ def parse_lambda_tags_from_arn(arn):
     ]
 
 
+MEMORY_ALLOCATED_RECORD_KEY = "memorySizeMB"
+INIT_DURATION_RECORD_KEY = "initDurationMs"
+BILLED_DURATION_RECORD_KEY = "billedDurationMs"
+RUNTIME_METRICS_BY_RECORD_KEY = {
+    # Except INIT_DURATION_RECORD_KEY which is handled separately
+    "durationMs": DURATION_METRIC_NAME,
+    BILLED_DURATION_RECORD_KEY: BILLED_DURATION_METRIC_NAME,
+    "maxMemoryUsedMB": MAX_MEMORY_USED_METRIC_NAME,
+}
+
+
+def parse_metrics_from_json_report_log(log_message):
+    if not log_message.startswith("{"):
+        return []
+
+    try:
+        body = json.loads(log_message)
+    except json.JSONDecodeError:
+        return []
+
+    stage = body.get("type", "")
+    record = body.get("record", {})
+    record_metrics = record.get("metrics", {})
+
+    if stage != "platform.report" or not record_metrics:
+        return []
+
+    metrics = []
+
+    for record_key, metric_name in RUNTIME_METRICS_BY_RECORD_KEY.items():
+        metric_point_value = record_metrics[record_key]
+
+        if metric_name in METRIC_ADJUSTMENT_FACTORS:
+            metric_point_value *= METRIC_ADJUSTMENT_FACTORS[metric_name]
+
+        metrics.append(
+            DatadogMetricPoint(
+                f"{ENHANCED_METRICS_NAMESPACE_PREFIX}.{metric_name}",
+                metric_point_value,
+            )
+        )
+
+    tags = [
+        f"{MEMORY_ALLOCATED_FIELD_NAME}:{record_metrics[MEMORY_ALLOCATED_RECORD_KEY]}"
+    ]
+
+    try:
```

Comment:
to have less try/except patterns

```suggestion
    init_duration = record_metrics.get(INIT_DURATION_RECORD_KEY)
    if init_duration:
        tags.append("cold_start:true")
        metrics.append(
            DatadogMetricPoint(
                f"{ENHANCED_METRICS_NAMESPACE_PREFIX}.{INIT_DURATION_METRIC_NAME}",
                init_duration * METRIC_ADJUSTMENT_FACTORS[INIT_DURATION_METRIC_NAME],
            )
        )
    else:
        tags.append("cold_start:false")
```

Reply:
Hm, for some reason, suggestion code doesn't have removed lines I'd expected it to have. I'll apply this change with the rest in one commit.